pkg/virt-api/rest/subresource.go

```diff
@@ -1364,24 +1369,51 @@ func (app *SubresourceAPIApp) fetchPersistentVolumeClaim(name string, namespace
 	return pvc, nil
 }
 
+func (app *SubresourceAPIApp) fetchCDIConfig() (*cdiv1.CDIConfig, *errors.StatusError) {
+	cdiConfig, err := app.virtCli.CdiClient().CdiV1beta1().CDIConfigs().Get(context.Background(), configName, k8smetav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, errors.NewInternalError(fmt.Errorf("unable to retrieve cdi config: %v", err))
+	}
+	return cdiConfig, nil
+}
+
 func (app *SubresourceAPIApp) validateMemoryDumpClaim(vmi *v1.VirtualMachineInstance, claimName, namespace string) *errors.StatusError {
 	pvc, err := app.fetchPersistentVolumeClaim(claimName, namespace)
 	if err != nil {
 		return err
 	}
-	if pvc.Spec.VolumeMode != nil && *pvc.Spec.VolumeMode == v12.PersistentVolumeBlock {
+	if k6ttypes.IsPVCBlock(pvc.Spec.VolumeMode) {
 		return errors.NewConflict(v1.Resource("persistentvolumeclaim"), claimName, fmt.Errorf(pvcVolumeModeErr))
 	}
 
+	if k6ttypes.IsReadOnlyAccessMode(pvc.Spec.AccessModes) {
+		return errors.NewConflict(v1.Resource("persistentvolumeclaim"), claimName, fmt.Errorf(pvcAccessModeErr))
+	}
+
 	pvcSize := pvc.Spec.Resources.Requests.Storage()
 	scaledPvcSize := resource.NewScaledQuantity(pvcSize.ScaledValue(resource.Kilo), resource.Kilo)
-	domain := vmi.Spec.Domain
-	vmiMemoryReq := domain.Resources.Requests.Memory()
-	memOverhead := resource.MustParse("100Mi")
-	expectedPvcSize := resource.NewScaledQuantity(vmiMemoryReq.ScaledValue(resource.Kilo), resource.Kilo)
-	expectedPvcSize.Add(memOverhead)
+
+	expectedMemoryDumpSize := kutil.CalcExpectedMemoryDumpSize(vmi)
```

Comment:
Does it make sense to combine this with the overhead calculation in its own function? the two are linked pretty tightly.

Reply:
I think the add of the overhead it relevant to any size for any pvc not only for specific memory dump